apis/v1alpha1/addon.go

```diff
@@ -31,8 +31,8 @@ type AddonSpec struct {
 	// of the request.
 	ClientRequestToken *string `json:"clientRequestToken,omitempty"`
 	// The name of the cluster to create the add-on for.
-	// +kubebuilder:validation:Required
-	ClusterName *string `json:"clusterName"`
+	ClusterName    *string                                  `json:"clusterName,omitempty"`
+	ClusterNameRef *ackv1alpha1.AWSResourceReferenceWrapper `json:"clusterNameRef,omitempty"`
```

Comment:
This got me thinking about resource reference fields and renaming of fields... What would be ideal is if we renamed the `ClusterName` field to `Cluster` and then the `ClusterRef` field would be better named...

Reply:
I don't think I agree. `ClusterName` clearly references a specific property of the resource, whereas `Cluster` would be ambiguous regarding whether it should be the ARN, the Name, or some other identifier. `ClusterNameRef` I think also works well, because it's very obviously going to be injecting the `Name` from the `Cluster` by `Ref[erence]`. Just my $0.02

Reply:
> I don't think I agree. `ClusterName` clearly references a specific property of the resource, whereas `Cluster` would be ambiguous regarding whether it should be the ARN, the Name, or some other identifier. `ClusterNameRef` I think also works well, because it's very obviously going to be injecting the `Name` from the `Cluster` by `Ref[erence]`. Just my $0.02

But it's not the cluster *name* you are referring to in the `ClusterNameRef` field. It's the Cluster.

Reply:
> > I don't think I agree. `ClusterName` clearly references a specific property of the resource, whereas `Cluster` would be ambiguous regarding whether it should be the ARN, the Name, or some other identifier. `ClusterNameRef` I think also works well, because it's very obviously going to be injecting the `Name` from the `Cluster` by `Ref[erence]`. Just my $0.02
> 
> But it's not the cluster _name_ you are referring to in the `ClusterNameRef` field. It's the Cluster.

True, but you are populating the `ClusterName` field by `Ref[erence]`. My thinking is that, say there is another field in the spec that also needs to reference a Cluster - you might want to reference a separate cluster for that part of the spec. It doesn't make a lot of sense for EKS Clusters, maybe, but I could see it being applicable for IAM or VPC where there are more many-to-one references.